runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/HttpProblem.java

```diff
@@ -21,7 +22,7 @@
  * Representation of RFC7807 Problem schema.
  */
 @Immutable
-public final class HttpProblem extends RuntimeException {
+public class HttpProblem extends RuntimeException {
```

Comment:
There are 2 things here that I'd like to express.

First of all why only `RuntimeException`?
Second: I'd prefer one of 2 options: 
* leave this class `final` or
*  *properly* extract interface and provide abstract class for convenience with some default implementation based on it - similar to Zalando lib.
Having this open (extendable) now will close possibility to avoid backward compatible changes in future - someone nearly for sure gonna extend this class it its code.

Reply:
I kind of agree on principle, but I personally don't see any benefit of such abstract class: can you elaborate on this? Why having abstract class is supposed to be easier to maintain? Once we need to make breaking changes for our extension we still can switch to using internal model (create `InternalProblem` class) without changing HttpProblem class public interface. Lets analyse all pros and cons of complicating types hierarchy.

Rule no 1 for me: let's keep it simple. For our needs I personally think we can go with a bit simple (maybe even 1 class) setup as is presented now. Zalando classes all depend on each other - interface has factory methods creating specific implementation instance etc - I'm not sure if I'm in fond of it too much :)

Reply:
Regarding `RuntimeException`: why not? ;) why `WebApplicationException extends RuntimeException`?